packages/preview/typographix-polytechnique-reports/0.1.3/cover.typ

```diff
@@ -0,0 +1,100 @@
+/***********************/
+/* TEMPLATE DEFINITION */
+/***********************/
+
+/* HANDLING DATE DISPLAY */
+
+#let translate_month(month) = {
+  // Construction mapping for months
+  let t = (:)
+  let fr-month-s = ("Janv.", "Févr.", "Mars", "Avr.", "Mai", "Juin",
+    "Juill.", "Août", "Sept.", "Oct.", "Nov.", "Déc.")
+  let fr-months-l = ("Janvier", "Février", "Mars", "Avril", "Mai", "Juin",
+    "Juillet", "Août", "Septembre", "Octobre", "Novembre", "Décembre")
+  for i in range(12) {
+    let idate = datetime(year: 0, month: i + 1, day: 1)
+    let ml = idate.display("[month repr:long]")
+    let ms = idate.display("[month repr:short]")
+    t.insert(ml, fr-months-l.at(i))
+    t.insert(ms, fr-month-s.at(i))
+  }
+
+  // Translating month
+  let fr_month = t.at(month)
+  fr_month
+}
+
+#let display-date(date, short-month) = {
+  context {
+    // Getting adapted month string
+    let repr = if short-month { "short" } else { "long" }
+    let month = date.display("[month repr:" + repr + "]")
+
+    // Translate if necessary
+    if text.lang == "fr" {
+      month = translate_month(month)
+    }
+
+  // Returns month and year
+  [#month #str(date.year())]
+  }
+}
+
+
+/* MAIN COVER DEFINITION */
+
+#let cover(title, author, date-start, date-end, subtitle: none, logo: none, short-month: false, logo-horizontal: true) = {
+  set page(background: move(dx: 0pt, dy: -13%, image("assets/armes.svg")))
+  set text(font: "New Computer Modern Sans", hyphenate: false, fill: rgb(1, 66, 106))
+  set align(center)
+
+  v(1.8fr)
+
+  set text(size: 24pt, weight: "bold")
+  upper(title)
+
+  v(1.5fr)
+
+  if subtitle != none {
+    set text(size: 20pt)
+    subtitle
+  }
+
+  v(1.5fr)
+  
+  set text(size: 18pt, weight: "regular")
+  display-date(date-start, short-month); [ \- ]; display-date(date-end, short-month)
+
+  image("assets/filet-court.svg")
+
+  set text(size: 16pt)
+  smallcaps(author)
+
+  v(1fr)
+
+  let logo-height = if (logo-horizontal) { 20mm } else { 30mm }
+  let path-logo-x = if (logo-horizontal) { "assets/logo-x-ip-paris.svg" } else { "assets/logo-x.svg" }
+
+  grid(
+    columns: (1fr, 1fr), align: center + horizon,
+    if (logo != none) { image(logo, height: logo-height) } else { smallcaps("Insert your logo") },
```

Comment:
If you expect `logo` to be a path to an image file, this won't work, as packages can only access files in their own directory, not in the user's project. What you should do instead, is directly take an `image` element as input, and let the user of your template call `image` with the correct path (you can still automatically apply height with a `set` rule).